test/fakeintake/server/server.go

```diff
@@ -46,9 +51,13 @@ type Server struct {
 // If the port is 0, a port number is automatically chosen
 func NewServer(options ...func(*Server)) *Server {
 	fi := &Server{
-		mu:           sync.RWMutex{},
-		payloadStore: map[string][]api.Payload{},
-		clock:        clock.New(),
+		mu:                 sync.RWMutex{},
+		payloadStore:       map[string][]api.Payload{},
+		payloadJsonStore:   map[string][]api.ParsedPayload{},
+		clock:              clock.New(),
+		logAggregator:      aggregator.NewLogAggregator(),
```

Comment:
As the number of routes to parse will increase over the time, I would create a dedicated struct that handles the parsing of the payloads and injects it into `NewServer`. As this field is not optional, I would not use `WithXXX`. What do you think?

Reply:
Yea I think it's a good idea to separate the parsing from the main ``Server`` struct.  Will do 👍 

>  As this field is not optional, I would not use WithXXX

I'm not sure I understand what you mean by that ?

Reply:
`NewServer` already has parameters using the [optional pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis). As the parsers are mandatory you should not use this pattern for the parsers.